docs/src/guides/languages/rust.md

```diff
@@ -9,4 +9,211 @@ tags:
 # :simple-rust: Rust
 <!-- markdownlint-enable single-h1 -->
 
-[TODO](https://github.com/input-output-hk/catalyst-ci/issues/78)
+<!-- cspell: words USERARCH TARGETARCH toolsets fmtchk stdcfgs rustfmt nextest testci testdocs -->
+
+## Introduction
+
+<!-- markdownlint-disable max-one-sentence-per-line -->
+!!! Tip
+    If you're just looking for a complete example,
+    [click here](https://github.com/input-output-hk/catalyst-ci/blob/master/examples/rust/Earthfile).
+    This guide will provide detailed instructions for how the example was built.
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+This guide will get you started with using the Catalyst CI to work with Rust based projects.
+
+To begin, clone the Catalyst CI repository:
+
+```bash
+git clone https://github.com/input-output-hk/catalyst-ci.git
+```
+
+Navigate to `examples/rust` to find a basic Rust project, with the `Earthfile` in it.
+This is the `Earthfile` we will be building in this guide.
+You can choose to either delete the file and start from scratch,
+or read the guide and follow along in the file.
+
+Also we will take a look how we are setup Rust projects and what configuration is used.
+
+## Building the Earthfile
+
+<!-- markdownlint-disable max-one-sentence-per-line -->
+!!! Note
+    The below sections will walk through building our `Earthfile` step-by-step.
+    In each section, only the fragments of the `Earthfile` relative to that section are displayed.
+    This means that, as you go through each section, you should be cumulatively building the `Earthfile`.
+    If you get stuck at any point, you can always take a look at the
+    [example](https://github.com/input-output-hk/catalyst-ci/blob/master/examples/rust/Earthfile).
+<!-- markdownlint-enable max-one-sentence-per-line -->
+
+### Prepare base builder
+
+```Earthfile
+VERSION 0.7
+
+# Set up our target toolchains, and copy our files.
+builder:
+    FROM ./../../earthly/rust+rust-base
+    
+    COPY --dir "*" .
```

Comment:
We specifically don't want this.  the reason CP_SRC exists is to make it easy to ONLY copy relevant source files.
We should not be copying files which are not part of the rust build.
For example cat-gateway has the full event-db schema in it.  None of that has anything to do with rust, copying it would cause the target to rebuild when the schema changes, but the rust code itself did not.
Or worse, if we add seed data.  None of which should trigger a rebuild of the rust target.

Reply:
I agree with your point to copy only needed files.
But `CP_SRC` actually does not solve this problem, it acts as the same as a regular `COPY` command but with less functionality which `COPY` has.